### PR TITLE
Update dependency Gradle to v8.1

### PR DESCRIPTION
**Pull request by 3flex**
https://docs.gradle.org/8.1/release-notes.html

### CODE REVIEW

File: build-logic/src/main/kotlin/packaging.gradle.kts [hunk 1]
```diff
@@ -4,10 +4,6 @@ plugins {
     signing
 }
 
-tasks.withType<Sign>().configureEach {
```
Comment (cortinico) [inline build-logic/src/main/kotlin/packaging.gradle.kts:7]: nice :)

**Review by cortinico — APPROVED** [review]

**Comment by cortinico** [comment]
Do we want this inside 1.23? Feels like yes

**Comment by 3flex** [comment]
I think so. It doesn't directly impact build artifacts; the configuration cache test change is just to align with Gradle's more correct behaviour in 8.1.

The build failure on Java 8 is an issue, I'll investigate. It might actually be a regression in Gradle.

**Comment by 3flex** [comment]
It's not a regression, but an issue that we didn't see previously because of a confluence of reasons that will take me too long to type up.

I've raised https://github.com/gradle/gradle/issues/24765 but in the meantime I've marked the `GenerateMavenPom` task incompatible with configuration cache, but only on Java 8. That will work around the issue and won't impact most contributors anyway since most will be running Gradle with more modern Java versions.

**Review by cortinico — APPROVED** [review]

**Pull request merged**